server/src/main/java/org/elasticsearch/index/mapper/ConstantFieldType.java

```diff
@@ -39,8 +39,9 @@ public final boolean isSearchable() {
         return true;
     }
 
+    // TODO make this final again once we've changed fieldDataBuilder to provide the context DataTierFieldType needs.
     @Override
-    public final boolean isAggregatable() {
+    public boolean isAggregatable() {
```

Comment:
This override is not needed. The logic in the base class is to check if `fieldDataBuilder` throws an `IllegalArgumentException`. So you just need to implement `fieldDataBuilder` to indicate that the field is aggregatable.